axelor-cash-management/src/main/java/com/axelor/apps/cash/management/service/InvoiceServiceManagementImpl.java

```diff
@@ -62,81 +66,100 @@ public InvoiceServiceManagementImpl(
-    LocalDate estimatedPaymentDate = invoice.getDueDate();
-    if (invoice.getPartner() != null && invoice.getPartner().getPaymentDelay() != null) {
-      estimatedPaymentDate =
-          estimatedPaymentDate.plusDays(invoice.getPartner().getPaymentDelay().intValue());
+  public Invoice computeEstimatedPaymentDate(Invoice invoice) {
+
+    if (CollectionUtils.isEmpty(invoice.getInvoiceTermList())) {
+      return invoice;
     }
-    if (invoice.getPaymentCondition() != null) {
-      if (invoice.getPaymentCondition().getTypeSelect() == PaymentConditionRepository.TYPE_NET) {
-        if (invoice.getPaymentCondition().getPeriodTypeSelect()
-            == PaymentConditionRepository.PERIOD_TYPE_DAYS) {
-          estimatedPaymentDate =
-              estimatedPaymentDate.plusDays(invoice.getPaymentCondition().getPaymentTime());
-        } else if (invoice.getPaymentCondition().getPeriodTypeSelect()
-            == PaymentConditionRepository.PERIOD_TYPE_MONTH) {
-          estimatedPaymentDate =
-              estimatedPaymentDate.plusMonths(invoice.getPaymentCondition().getPaymentTime());
-        }
-      } else if (invoice.getPaymentCondition().getTypeSelect()
-          == PaymentConditionRepository.TYPE_END_OF_MONTH_N_DAYS) {
-        estimatedPaymentDate =
-            estimatedPaymentDate.withDayOfMonth(estimatedPaymentDate.lengthOfMonth());
-        if (invoice.getPaymentCondition().getPeriodTypeSelect()
-            == PaymentConditionRepository.PERIOD_TYPE_DAYS) {
-          estimatedPaymentDate =
-              estimatedPaymentDate.plusDays(invoice.getPaymentCondition().getPaymentTime());
-        } else if (invoice.getPaymentCondition().getPeriodTypeSelect()
-            == PaymentConditionRepository.PERIOD_TYPE_MONTH) {
-          estimatedPaymentDate =
-              estimatedPaymentDate.plusMonths(invoice.getPaymentCondition().getPaymentTime());
-        }
-      } else if (invoice.getPaymentCondition().getTypeSelect()
-          == PaymentConditionRepository.TYPE_N_DAYS_END_OF_MONTH) {
-        if (invoice.getPaymentCondition().getPeriodTypeSelect()
-            == PaymentConditionRepository.PERIOD_TYPE_DAYS) {
-          estimatedPaymentDate =
-              estimatedPaymentDate.plusDays(invoice.getPaymentCondition().getPaymentTime());
-        } else if (invoice.getPaymentCondition().getPeriodTypeSelect()
-            == PaymentConditionRepository.PERIOD_TYPE_MONTH) {
+    if (invoice.getPartner() != null && invoice.getPartner().getPaymentDelay() != null) {
+
+      int paymentDelay = invoice.getPartner().getPaymentDelay().intValue();
+
+      for (InvoiceTerm invoiceTerm : invoice.getInvoiceTermList()) {
+        invoiceTerm.setEstimatedPaymentDate(invoiceTerm.getDueDate().plusDays(paymentDelay));
+      }
+    } else {
+      for (InvoiceTerm invoiceTerm : invoice.getInvoiceTermList()) {
+        LocalDate estimatedPaymentDate = invoiceTerm.getDueDate();
+
+        if (invoiceTerm.getPaymentConditionLine().getTypeSelect()
```

Comment:
payment condition line is not required in invoice term domain, so I think this can throw a NPE.